lightning/src/util/ser_macros.rs

```diff
@@ -184,6 +184,9 @@ macro_rules! decode_tlv {
 	($reader: expr, $field: ident, (option: $trait: ident $(, $read_arg: expr)?)) => {{
 		$field = Some($trait::read(&mut $reader $(, $read_arg)*)?);
 	}};
+	($reader: expr, $field: ident, (option, encoding: ($fieldty: ty, $encoding: ident, $encoder:ty))) => {{
+		decode_tlv!($reader, $field, (option, encoding: ($fieldty, $encoding)));
```

Comment:
Looks like I'll need to rebase to account for #1823. Let me know when it's good to squash and rebase.